app/src/test/java/org/groundplatform/android/ui/datacollection/tasks/polygon/DrawAreaTaskViewModelTest.kt

```diff
@@ -222,6 +230,29 @@ class DrawAreaTaskViewModelTest : BaseHiltTest() {
     assertGeometry(4, isLineString = true)
   }
 
+  @Test
+  fun `redoLastVertex re-adds last vertex`() {
+    updateLastVertexAndAdd(COORDINATE_1)
+    updateLastVertexAndAdd(COORDINATE_2)
+
+    viewModel.removeLastVertex()
+    assertGeometry(2, isLineString = true)
+
+    viewModel.redoLastVertex()
+    assertGeometry(3, isLineString = true)
+  }
+
+  @Test
+  fun `redoLastVertex when redo stack is empty`() {
+    updateLastVertexAndAdd(COORDINATE_1)
+
+    viewModel.redoLastVertex()
+    assertThat(viewModel.redoVertexStack).isEqualTo(emptyList<Coordinates>())
+
+    viewModel.redoLastVertex()
+    assertThat(timberRule.logs, hasItem(endsWith("redoVertexStack is already empty")))
```

Comment:
We shouldn't be checking the log message in unit tests. Please remove

Reply:
<img width="517" alt="Screenshot 2025-05-21 at 5 25 18 PM" src="https://github.com/user-attachments/assets/8d716bd5-53ee-43aa-85b4-c8fe6d5b5332" />

for code coverage, should I remove or keep?